draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -124,30 +124,38 @@ In the remainder of the document, it is assumed that each browser execution cont
 ICE Candidate Gathering {#gathering}
 ----------------------------
 
-For any host ICE candidate gathered by a browsing context as part of {{RFC8445}} section 5.1.1, obfuscation of the candidate is done as follows:
+For any host candidate gathered by an ICE agent as part of {{RFC8445}} section 5.1.1, obfuscation of the candidate is done as follows:
 
-1. Check whether the context ICE agent registered a name resolving to the ICE host candidate IP address.
+1. Check whether there is a usable registered mDNS hostname resolving to the ICE host candidate's IP address.
 
-2. If the ICE agent registered the name, replace the IP address of the ICE host candidate with the name with ".local" appended to it. Expose the candidate and abort these steps.
+2. If there is a registered a hostname, replace the IP address of the ICE host candidate with the hostname. Expose the candidate and abort these steps.
 
-3. Generate a random unique name, typically a version 4 UUID as defined in {{RFC4122}}.
+3. Generate a unique name. The unique name SHOULD consist of a version 4 UUID as defined in {{RFC4122}}. The name MUST NOT contain any ".".
 
-4. Register the unique name using Multicast DNS.
+4. Append ".local" to the generated name, creating the mDNS hostname.
 
-5. If registering of the unique name fails, abort these steps. The candidate is not exposed.
+5. Register the candidate's mDNS hostname using Multicast DNS.
 
-6. Store the name and its related IP address in the ICE agent for future reuse.
+6. If registering of the mDNS hostname fails, abort these steps. The candidate is not exposed.
 
-7. Replace the IP address of the ICE host candidate with the name with ".local" appended to it. Expose the candidate.
+7. Store the mDNS hostname and its related IP address in the ICE agent for future reuse.
+
+8. Replace the IP address of the ICE host candidate with its mDNS hostname. Expose the candidate.
+
+The use of any registered mDNS hostnames MUST be limited in time and/or scope. Indefinitely reusing the same mDNS hostname candidate would provide applications an even more reliable tracking mechanism than the private IP addresses that this specification is designed to hide. The use of an mDNS name can be restricted to a browser origin, a peer-connection or any other context the implementation chooses.
```

Comment:
As discussed today, I think we want to take a specific position here, and say that mDNS hostnames SHOULD be scoped by origin, and have the lifetime of the page.

Reply:
Changed to MUST.